src/AxelarGateway.sol

```diff
@@ -43,6 +49,45 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         _;
     }
 
+    /******************\
+    |* Public Methods *|
+    \******************/
+
+    function sendToken(
+        uint256 destinationChain,
+        string memory symbol,
+        uint256 amount
+    ) external returns (bool) {
+        address tokenAddress = tokenAddresses(symbol);
+        require(tokenAddress != address(0), 'TOKEN_NOT_EXIST');
+
+        TokenType tokenType = _getTokenType(symbol);
+        bytes memory tokenCallData;
+        bytes32 salt;
+
+        if (tokenType == TokenType.External) {
+            tokenCallData = abi.encodeWithSelector(IERC20.transferFrom.selector, msg.sender, address(this), amount);
+        } else if (tokenType == TokenType.InternalBurnableFrom) {
+            tokenCallData = abi.encodeWithSelector(BurnableMintableCappedERC20.burnFrom.selector, msg.sender, amount);
+        } else if (tokenType == TokenType.InternalBurnable) {
+            salt = keccak256(abi.encodePacked(destinationChain, msg.sender, tokenAddress));
```

Comment:
At this point, by bother creating a different deposit address for each user? Why not just have everyone to share the same as 
```
address depositAddress = BurnableMintableCappedERC20(tokenAddress).depositAddress(bytes32(0));
```

Reply:
I think you are right, and this will always work. But still feels dangerous 😄